packages/esm-patient-registration-app/src/widgets/edit-patient-details-button.component.tsx

```diff
@@ -1,20 +1,40 @@
 import React from 'react';
-import { ConfigurableLink } from '@openmrs/esm-framework';
+import { navigate } from '@openmrs/esm-framework';
 import { useTranslation } from 'react-i18next';
 import styles from './edit-patient-details-button.scss';
 
-export default function EditPatientDetailsButton({ patientUuid }) {
+interface EditPatientDetailsButtonProps {
+  closePatientSearchResultsPanel?: () => void;
+  patientUuid: string;
+}
+
+const EditPatientDetailsButton: React.FC<EditPatientDetailsButtonProps> = ({
+  patientUuid,
+  closePatientSearchResultsPanel,
```

Comment:
This button seems to know too much about its context. I think we should establish a pattern to deal with this kind of situation, where we have a function prop that should be called on click/transition. Maybe we should call this prop `onTransition`? Or `afterClick`?